python/turbine_models/gen_external_params/gen_external_params.py

```diff
@@ -73,34 +55,105 @@ def forward(self, x):
     return all_weights
 
 
-if __name__ == "__main__":
-    args = parser.parse_args()
+def gen_external_params(
+    hf_model_name: str = "meta-llama/Llama-2-7b-chat-hf",
+    quantization: str = "int4",
+    weight_path: str = "",
+    hf_auth_token: str = None,
+    precision: str = "f16",
+):
+    """
+    Main function to run the model quantization and saving process.
+
+    :param hf_model_name: The Hugging Face model name ID.
+    :param quantization: Type of quantization to apply ('int4' or 'int8').
```

Comment:
right now int4 and int8 are the only ones that do quantization otherwise we don't quantize. Might be better to have it be None, int4 or int8 and provide an error if it's neither of those to avoid typo confusion

Reply:
Will use unquantized instead of none because it would be easier to generate a safe_name